msprime/trees.py

```diff
@@ -650,6 +676,12 @@ def draw(
         :param bool show_mutation_labels: If True, show labels for mutations.
         :param bool show_internal_node_labels: If True, show labels for internal nodes.
         :param bool show_leaf_node_labels: If True, show labels for leaf nodes.
+        :param list branch_colours: A list giving colours for edges in the tree.
+            If this is an array such as ['red','blue'], the indexes of the array
+            correspond to the node whose edges are coloured. If a dict, the keys
+            give the node numbers for each colour. Unspecified edges are coloured  
```

Comment:
It's not hard. Suppose I want to colour the clade containing nodes 0 and 1:
```python
for t in ts.trees():
     branch_colours = {u: "red" for u in  t.nodes(t.mrca(0, 1))}
     t.draw(branch_colours=branch_colours)
```
There's no way we can predict all possible ways in which people might want to colour upwards form the leaves, and I **strongly** believe that it's a bad idea to try. This general interface allows the client full power to colour branches any way they want.